internal/requests/auth_code_request_test.go

```diff
@@ -9,62 +9,92 @@ import (
 	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/msalbase"
 )
 
-var testAuthorityEndpoints = msalbase.CreateAuthorityEndpoints("https://login.microsoftonline.com/v2.0/authorize",
-	"https://login.microsoftonline.com/v2.0/token",
-	"https://login.microsoftonline.com/v2.0",
-	"https://login.microsoftonline.com")
-var testAuthorityInfo, err = msalbase.CreateAuthorityInfoFromAuthorityURI("https://login.microsoftonline.com/v2.0/", true)
-var testAuthParams = msalbase.CreateAuthParametersInternal("clientID", testAuthorityInfo)
+func createTestAuthorityInfo() msalbase.AuthorityInfo {
+	info, err := msalbase.CreateAuthorityInfoFromAuthorityURI("https://login.microsoftonline.com/v2.0/", true)
+	if err != nil {
+		panic(err)
+	}
+	return info
+}
+
+func createTDR() TenantDiscoveryResponse {
+	return TenantDiscoveryResponse{
+		AuthorizationEndpoint: "https://login.microsoftonline.com/v2.0/authorize",
+		TokenEndpoint:         "https://login.microsoftonline.com/v2.0/token",
+		Issuer:                "https://login.microsoftonline.com/v2.0",
+	}
+}
+
+func createACRTestParams() msalbase.AuthParametersInternal {
+	tdr := createTDR()
+	params := msalbase.CreateAuthParametersInternal("clientID", createTestAuthorityInfo())
+	// TODO(msal expert): This is the only change here that is an actual change of the test. This is
+	// also changed in other mock tests. Found that calls to methods like AuthCodeRequest.Execute()
+	// would set these Endpoints. But for some reason, none of the mocks expected it.
+	// I'm not sure why this worked (another reason for not using mocks), but I am
+	// pretty sure the mock call to GetAccessTokenFromAuthCode should have seen these.
+	// So I fixed it here, but someone should verify I'm not doing something bad.
```

Comment:
It looks like the Mocks where only used to set return values, and not actually verify all the calls. I'm not seeing any calls to `AssertExpectations`

Reply:
That isn't the case with all mocks, but certainly the ones here.  The mocks should be replaced and the code should test the outcomes.   

However, this CL doesn't cover those types of changes, just changes enough to keep everything working.